src/EventTickets/resources/admin/components/EventFormModal/index.tsx

```diff
@@ -0,0 +1,157 @@
+import {SubmitHandler, useForm} from 'react-hook-form';
+import {__} from '@wordpress/i18n';
+import styles from './EventFormModal.module.scss';
+import FormModal from '../FormModal';
+import EventTicketsApi from '../api';
+
+/**
+ * Get the next sharp hour
+ *
+ * @unreleased
+ */
+const getNextSharpHour = (hoursToAdd: number) => {
+    const date = new Date();
+    date.setHours(date.getHours() + hoursToAdd, 0, 0, 0);
+
+    return date;
+};
+
+/**
+ * Format a given date to be used in datetime inputs
+ *
+ * @unreleased
+ */
+const getDateString = (date: Date) => {
+    if (!date) {
+        return;
+    }
+
+    const offsetInMs = date.getTimezoneOffset() * 60 * 1000;
+    const dateWithOffset = new Date(date.getTime() - offsetInMs);
+
+    return dateWithOffset.toISOString().slice(0, -8);
+};
+
+/**
+ * Event Form Modal component
+ *
+ * @unreleased
+ */
+export default function EventFormModal({isOpen, handleClose, apiSettings, title, event}: EventModalProps) {
+    const API = new EventTicketsApi(apiSettings);
+
+    const {
+        register,
+        handleSubmit,
+        formState: {errors, isDirty},
+    } = useForm<Inputs>({
+        defaultValues: {
+            title: event?.title ?? '',
+            description: event?.description ?? '',
+            startDateTime: getDateString(
+                event?.startDateTime?.date ? new Date(event?.startDateTime?.date) : getNextSharpHour(1)
+            ),
+            endDateTime: getDateString(
+                event?.endDateTime?.date ? new Date(event?.endDateTime?.date) : getNextSharpHour(2)
+            ),
+        },
+    });
+
+    const onSubmit: SubmitHandler<Inputs> = async (inputs) => {
+        try {
+            inputs.startDateTime += ':00';
+            inputs.endDateTime += ':00';
```

Comment:
It feels like a red flag to be appending a specific format, like `:00` to date time values. Is there a better way to handle this?

Reply:
Proposal: 502c7d0fecbb1415c5124d327cad3ad4c79b8557

Reply:
Could we use [Date.prototype.toISOString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString) to simplify this formatting? Create a `new Date` object from the string and then format to an ISO date string.

Reply:
When I use `toISOString`, it converts my time to UTC instead of respecting my current timezone, resulting in a wrong time being saved.

Reply:
I think is important to note that this behavior is due to the input we are using. Whenever we implement a new DateTime selector, this may become not relevant anymore.

Reply:
What about using `getDateString()` from above? Would that work?

Reply:
It seems to work but I had to do one change on that method, keep the seconds in the string. See 5450ecb53233cbeb6d8ab652f4d2da69ae22ea25

Reply:
Okay - that works.

Lastly, since we are using a magic number to slice the date string, can you leave a comment that explains what the magic number represents?

Better yet, we could extract a declarative function, something like, `removeTimezoneFromDateString()`, so that the function name describes the intention without needing to add a comment.

Reply:
Extracted: 3ea38282880161d01cdbb626aad01ea0efae2988